portfolio/src/main/webapp/calendarApi.js

```diff
@@ -0,0 +1,298 @@
+// require(['dotenv']).config({path: './.env'});
+
+var GoogleAuth;
+
+// Scopes for API access to Google Calendar
+const SCOPE_READ_ONLY = 'https://www.googleapis.com/auth/calendar.readonly';
+const SCOPE_READ_WRITE = 'https://www.googleapis.com/auth/calendar';
+
+// For the discovery document for Google Calendar API.
+const DISCOVERY_URL =
+    'https://www.googleapis.com/discovery/v1/apis/calendar/v3/rest';
+
+const CLIENT_ID =
+    '499747085593-hvi6n4kdrbbfvcuo1c9a9tu9oaf62cr2.apps.googleusercontent.com';
+
+// Constants for error codes during OAuth process.
+const ERROR_CODE_POPUP_CLOSED = 'popup_closed_by_user';
```

Comment:
Consider using Javascript object. Maybe it can be its own file as well, like errorType.js. It can export an object like

{
  'popup_closed_by_user': 'popup_closed_by_user',
  ...
}

Reply:
I moved the constants into an Object in this file. I am having issues with the "require" keyword, similar to the one I have when I try to store API keys in another variable, so for right now I left the error code object in the same file.